mmpose/core/post_processing/matching.py

```diff
@@ -0,0 +1,297 @@
+import numpy as np
+import torch
+from scipy.optimize import linear_sum_assignment
+
+sigmas = np.array([
+    .26, .25, .25, .35, .35, .79, .79, .72, .72, .62, .62, 1.07, 1.07, .87,
+    .87, .89, .89
+])
+
+
+def candidate_reselect(bboxes, pose_preds, num_joints, img, box_scores,
+                       in_vis_thre):
+    """get final result with group and match.
+
+    Note:
+        num_person: N
+    Args:
+        bboxes (torch.Tensor[N,4]): bbox(x1,y1,x2,y2).
+        pose_preds (dict): candidate keypoints information.
+            pose_preds[i][j] (list): i-th person j-th joint 's canditate
+                    keypoints information.
+        num_joints (int): number of keypoints.
+        img (int): image_id.
+        box_scores (torch.Tensor[N,1]): bbox score.
+        in_vis_thre (float): threshold.
+    Returns:
+        final_result (list): predicted in the image.
+    """
+
+    # Group same keypointns together
+    kp_groups = _grouping(bboxes, pose_preds, num_joints, box_scores)
+
+    # Generate Matrix
+    human_num = len(pose_preds.keys())
+    costMatrix = []
+
+    for k in range(num_joints):
+        kp_group = kp_groups[k]
+        joint_num = len(kp_group.keys())
+        costMatrix.append(np.zeros((human_num, joint_num)))
+
+    for n, person in pose_preds.items():
```

Comment:
1. what's the difference between person and human?
2. either use `n` or use `h_id` in this loop, not both of them